Source/WebCore/rendering/RenderCounter.cpp

```diff
@@ -466,26 +466,25 @@ String RenderCounter::originalText() const
     RefPtr child = m_counterNode.get();
     int value = child->actsAsReset() ? child->value() : child->countInParent();
 
-    auto counterText = [](const ListStyleType& styleType, int value, CSSCounterStyle* counterStyle) {
+    auto counterText = [](const ListStyleType& styleType, int value, CSSCounterStyle* counterStyle, TextDirection textDirection) {
         if (styleType.type == ListStyleType::Type::None)
             return emptyString();
 
         if (styleType.type == ListStyleType::Type::CounterStyle) {
             ASSERT(counterStyle);
-            return counterStyle->text(value);
+            return counterStyle->text(value, textDirection);
         }
 
         ASSERT_NOT_REACHED();
         return emptyString();
     };
     auto counterStyle = this->counterStyle();
-    String text = counterText(m_counter.listStyleType(), value, counterStyle.get());
-
+    auto text = counterText(m_counter.listStyleType(), value, counterStyle.get(), style().direction());
     if (!m_counter.separator().isNull()) {
         if (!child->actsAsReset())
             child = child->parent();
         while (CounterNode* parent = child->parent()) {
-            text = counterText(m_counter.listStyleType(), child->countInParent(), counterStyle.get())
+            text = counterText(m_counter.listStyleType(), child->countInParent(), counterStyle.get(), textDirection)
```

Comment:
Why is this using textDirection but the line above is using style().direction ?

Reply:
ops, I've messed this up when I've merged the 2 patches. Let me fix that.